src/journal/bsod.c

```diff
@@ -0,0 +1,244 @@
+/* SPDX-License-Identifier: LPGL-2.1-or-later */
+
+#include <errno.h>
+#include <getopt.h>
+#include <linux/vt.h>
+#include <sys/ioctl.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "build.h"
+#include "fd-util.h"
+#include "io-util.h"
+#include "log.h"
+#include "main-func.h"
+#include "qrcode-util.h"
+#include "sd-id128.h"
+#include "sd-journal.h"
+#include "sysctl-util.h"
+#include "terminal-util.h"
+
+static void help(void) {
+        printf("%s\n\n"
+               "filters the journal to fetch the first message from the\n"
+               " current boot with an emergency log level.\n\n"
+               "   -h --help            Show this help\n"
+               "      --version         Show package version\n",
+               program_invocation_short_name);
+}
+
+static int acquire_first_emergency_log_message(char **ret) {
+        _cleanup_(sd_journal_closep) sd_journal *j = NULL;
+        _cleanup_free_ char *message = NULL;
+
+        const void *d;
+        size_t l;
+        sd_id128_t boot_id;
+        char boot_id_filter[STRLEN("_BOOT_ID=") + SD_ID128_STRING_MAX];
+        int r;
+
```

Comment:
assert(ret)

Reply:
> assert(ret)

This would mean, we have to always initialize any pointer whose address we pass to this function?

Reply:
no, only the return storage